ios/engine/KMEI/KeymanEngine/resources/Keyman.bundle/Contents/Resources/ios-host.js

```diff
@@ -276,33 +276,21 @@ function doResetContext() {
     keyman.resetContext();
 }
 
-function setCursorRange(pos, length) {
-    var context = keyman.context;
-
-    //console.log('setCursorRange('+pos+', '+length+')');
-
-    var start = pos;
-    var end = pos + length;
-
-    if(context.selStart != start || context.selEnd != end) {
-      keyman.context.setSelection(start, end);
-      keyman.resetContext();
-    }
-}
-
-function setKeymanVal(text) {
-    //console.log('setKeymanVal('+JSON.stringify(text)+')');
-
+function setKeymanContext(text, doSync, pos, length) {
+    // console.log(`setKeymanContext(${JSON.stringify(text)}, ${doSync}, ${pos}, ${length})`);
     if(text == undefined) {
         text = '';
     }
 
-    if(keyman.context.getText() != text) {
-        keyman.context.setText(text);
+    var start = pos;
+    var end = pos + length;
+    // undefined + <number> => NaN.
+    end = isNaN(end) ? undefined : end;
+
+    const shouldReset = keyman.context.updateContext(text, start, end);
+    if(!doSync || shouldReset) {
         keyman.resetContext();
     }
-
-    return keyman.context.getText();
```

Comment:
Was this never used?

Reply:
Yes.  Here's the only thing calling it:

https://github.com/keymanapp/keyman/blob/66142be3930c24f996939f7496a377f23223e3c3/ios/engine/KMEI/KeymanEngine/Classes/Keyboard/KeymanWebViewController.swift#L249

Note how nothing is checking for a result.  To be extra clear here, that would be done by the second argument if one were provided.

https://developer.apple.com/documentation/webkit/wkwebview/1415017-evaluatejavascript

No callback means no need for a return value.